### PR TITLE
fix(artifacts): Increase number/duration of retries for artifact metadata

### PR DESCRIPTION
**Pull request by luispollo**
Doubles the number of retries for artifact metadata and spaces them out over a longer period to improve our chances. Also logs retries dues to empty results from Igor/Rocket.

### CODE REVIEW

**Review by lorin — APPROVED** [review]
Change looks good to me, but could you say a little more about the motivation for this change? What's the issue that the retries are working around?

**Comment by luispollo** [comment]
> Change looks good to me, but could you say a little more about the motivation for this change? What's the issue that the retries are working around?

Yes, sorry, I should have included more context in the PR description. We currently attempt to retrieve metadata only once, when a new version of an artifact is detected, which is why we have retries in this code path. However, we've found that if there's ever anything that causes the downstream CI system (Rocket) to take longer to index the build metadata, or become unavailable for any reason, for longer than we're retrying, than we end up with missing metadata. This change is not a fail-proof way of ensuring we do, but is an attempt at improving the odds. If we still fail, @gal-yardeni is working on a scheduled batch job to backfill the missing metadata (see #1549).

File: keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt [hunk 1]
```diff
@@ -95,18 +95,23 @@ class ArtifactMetadataService(
     val retry = Retry.of(
       "get artifact metadata",
       RetryConfig.custom<List<Build>?>()
-        .maxAttempts(10)
-        .waitDuration(Duration.ofMillis(1000))
-        .retryOnResult{
-          result -> result.isNullOrEmpty()
+        // retry 20 times over a total of 90 seconds
```
Comment (luispollo) [inline keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt:98]: @emjburns Does a minute and a half worth of additional waiting sound acceptable to you (worst case scenario)?

**Review by gal-yardeni — APPROVED** [review]
LGTM!

**Pull request merged**